packages/itwin/measure-tools/src/api/DrawingTypeDataCache.ts

```diff
@@ -7,38 +7,71 @@ import type { IModelConnection } from "@itwin/core-frontend";
 import { IModelApp } from "@itwin/core-frontend";
 import { SheetMeasurementsHelper } from "./SheetMeasurementHelper";
 
-export class DrawingDataCache {
+class DrawingDataCache {
+
+  // Goes from viewed model to drawing types
+  private _drawingTypeCache: Map<string, SheetMeasurementsHelper.DrawingTypeData[]>;
 
-  private _drawingTypeCache: SheetMeasurementsHelper.DrawingTypeData[];
   private _sheetChangeListener: VoidFunction[] = [];
 
-  public get drawingtypes(): ReadonlyArray<SheetMeasurementsHelper.DrawingTypeData> {
-    return this._drawingTypeCache;
+  public getDrawingtypes(viewedModelID: string): Readonly<SheetMeasurementsHelper.DrawingTypeData[]> {
+    const result = this._drawingTypeCache.get(viewedModelID);
+    return result ?? [];
+  }
+
+  public constructor(imodel: IModelConnection) {
+    this._drawingTypeCache = new Map<string, SheetMeasurementsHelper.DrawingTypeData[]>();
+    this.updateDrawingTypeCache(imodel);
   }
 
-  public constructor() {
-    this._drawingTypeCache = [];
+  public destructor() {
```

Comment:
There's already such thing as implementing the `IDisposable` interface, maybe you'll want to rename this to `public dispose(): void`